specification/metrics/semantic_conventions/system-metrics.md

```diff
@@ -39,6 +39,7 @@ instruments not explicitly defined in the specification.
 |                        |                                                                                                          |       |                                                   |            | cpu              | CPU number [0..n-1]                 |
 | system.cpu.utilization | Difference in system.cpu.time since the last measurement, divided by the elapsed time and number of CPUs | 1     | Gauge                                             | Double     | state            | idle, user, system, interrupt, etc. |
 |                        |                                                                                                          |       |                                                   |            | cpu              | CPU number (0..n)                   |
+| system.cpu.count       | Reports the and number of available CPUs - may changes over time                                         | 1     | UpDownCounters                                    | Int64      | device           | (identifier e.g. `machine-id`)      |
```

Comment:
@dmitryax, @mx-psi you may want to have a look.

Reply:
While I understand the practical benefit of having this as a metric (as mentioned on https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22099#issuecomment-1581682604), I still feel like if we abide strictly by the data model this is more of a resource attribute: a fixed (at least, as fixed as `os.version` or `host.image.version` can be) property of a resource that generates telemetry. I don't have a strong opinion on whether practical concerns should trump the data model purity argument, but I think we should still consider both.

For Datadog's use case a resource attribute would be better, but I don't think our use case is very common (we intend to map this information to a vendor-specific payload at the exporter level and a resource attribute allows for statelessness in deployments while if we have this information as metrics we need all metrics to end up on the same payload).

Reply:
The discussion on open-telemetry/opentelemetry-specification/pull/2384 is also relevant. This comment https://github.com/open-telemetry/opentelemetry-specification/pull/2384#issuecomment-1056063450 makes me think that my thumb rule of "if it has a fixed value over time then it's not a metric" makes sense.

@trask @bogdandrutu @tigrannajaryan, any thoughts on this PR and the metric vs. resource attribute discussion?

Reply:
> if it has a fixed value over time then it's not a metric

But is it a fixed value? My I am not an expert on that field, but my understanding is that it can change.
At least this [random VMware article](https://www.bdrsuite.com/blog/what-is-vmware-hot-add-hot-plug-how-will-it-impact-numa/) claims it too. ^^

Reply:
I'll also share this scenario as public clouds allow you to resize an instance the cpu count can certainly change over the lifetime of a system: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-resize.html

Reply:
Sounds like I am the only one arguing for resource attribute, I suppose we can mark this discussion as solved

Reply:
> I'll also share this scenario as public clouds allow you to resize an instance the cpu count can certainly change over the lifetime of a system: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-resize.html

Sorry to comment on a closed PR, but that user guide clearly says that you need to change your instance type to change the number of CPUs. This means turning off the old instance and start up a new one, including IP address change.

Similar to @mx-psi, I'd argue for CPU count being a resource attribute. Is it possible to have CPU count as metric and resource attribute?